src/controllers/engine/controllerengine.cpp

```diff
@@ -74,50 +74,72 @@ Purpose: Calls the same method on a list of JS Objects
 Input:   -
 Output:  -
 -------- ------------------------------------------------------ */
-void ControllerEngine::callFunctionOnObjects(QList<QString> scriptFunctionPrefixes,
-                                             const QString& function, QScriptValueList args) {
-    VERIFY_OR_DEBUG_ASSERT(m_pEngine) {
-        return;
+bool ControllerEngine::callFunctionOnObjects(QList<QString> scriptFunctionPrefixes,
+        const QString& function,
+        QJSValueList args,
+        bool bFatalError) {
+    VERIFY_OR_DEBUG_ASSERT(m_pScriptEngine) {
+        return false;
     }
 
-    const QScriptValue global = m_pEngine->globalObject();
+    const QJSValue global = m_pScriptEngine->globalObject();
 
+    bool success = true;
     for (const QString& prefixName : scriptFunctionPrefixes) {
-        QScriptValue prefix = global.property(prefixName);
-        if (!prefix.isValid() || !prefix.isObject()) {
+        QJSValue prefix = global.property(prefixName);
+        if (!prefix.isObject()) {
             qWarning() << "ControllerEngine: No" << prefixName << "object in script";
             continue;
         }
 
-        QScriptValue init = prefix.property(function);
-        if (!init.isValid() || !init.isFunction()) {
+        QJSValue init = prefix.property(function);
+        if (!init.isCallable()) {
             qWarning() << "ControllerEngine:" << prefixName << "has no" << function << " method";
             continue;
         }
         controllerDebug("ControllerEngine: Executing" << prefixName << "." << function);
-        init.call(prefix, args);
+        QJSValue result = init.callWithInstance(prefix, args);
+        if (result.isError()) {
+            showScriptExceptionDialog(result, bFatalError);
+            success = false;
+        }
     }
+    return success;
+}
+
+QJSValue ControllerEngine::byteArrayToScriptValue(const QByteArray byteArray) {
+    // The QJSEngine converts the QByteArray to an ArrayBuffer object.
+    QJSValue arrayBuffer = m_pScriptEngine->toScriptValue(byteArray);
+    // We convert the ArrayBuffer to a Uint8 typed array so we can access its bytes
+    // with the [] operator.
+    QJSValueList args;
+    args << arrayBuffer;
+    QJSValue result = m_byteArrayToScriptValueJSFunction.call(args);
```

Comment:
m_byteArrayToScriptValueJSFunction.call(QJSValueList{arrayBuffer})

Reply:
I disagree. The result needs to be checked if it is an error on the next line. IMO executing actions in `if` conditions is an antipattern that makes for confusing code.